astropy/utils/masked/function_helpers.py

```diff
@@ -673,37 +778,127 @@ def _masked_quantile(a, q, axis=None, out=None, **kwargs):
     return result
 
 
-@dispatched_function
-def quantile(a, q, axis=None, out=None, **kwargs):
+if not NUMPY_LT_2_0:
+    _quantile_is_valid = np.lib._function_base_impl._quantile_is_valid
```

Comment:
I think I'd still write this as
```
from numpy.lib._function_base_impl import _quantile_is_valid, _check_interpolation_as_method
```
etc.

Actually, might as well get `_ureduce` as well, since that's used in the actual implementations.

Reply:
good thinking, done !